packages/shared/sierra-client/tests/email-verification-notes.test.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  addVerificationNote,
+  deleteNonCurrentVerificationNotes,
+  verifiedEmails,
+} from '../src/email-verification-notes';
+import { VarField, varFieldTags } from '../src/patron';
+
+const notesToVarFields = (...notes: string[]): VarField[] =>
+  notes.map((content) => ({
+    fieldTag: varFieldTags.notes,
+    content,
+  }));
+
+describe('email verification notes', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+    jest.setSystemTime(new Date('2022-02-22'));
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  describe('addVerificationNote', () => {
```

Comment:
What if we add a verification note and there's already one there? What if the existing note has the same email address? A different email address?

Reply:
I didn't test for these because it just does what it says - it adds one and doesn't care about the others

Reply:
\*nod* – I think that's being covered in tests in a subsequent PR.

Reply:
Okay, no, I'm confused again.

I guess maybe this is a design question I want to clarify: when a user changes their email address, do we scrub the previous verification notes, or do we keep them? It looks like maybe we only scrub them on their next login, is that correct?

Reply:
Your understanding is correct, and answering this question is the sole reason that this PR took so long to come up with. 

If you scrub the previous verification note when the user changes their email address, then your user becomes indistinguishable from a user who's appropriate for "implicit verification" - so they can verify themselves just by logging in again, which is wrong. 

However, now that I think about it again having done some more work on this, I think there might be another way to do it. 
- When a user changes their email address, do nothing to their notes (we can see that they're not verified because their email does not match their verification note)
- When a user verifies their email address, add the new verification note and delete the old one.
- When a user logs in, add the (implicit) verification note if they have _no_ verification notes.

I do remember trying this and discounting it, but perhaps something else changed after I did this - not sure. Can you see any problems with it?

Reply:
\*nod* Yup, I think I was suggesting closer to the second flow – when I said "changes their email address", I really meant "verifies their email address after changing it". The second flow sounds fine to me, but now I'm trying to think if there's a hole.

The only wrinkle I can think of (but it applies both way) is a well-meaning staff member deleting a verification note for an old email address, and getting us into the implicit flow. But that seems quite unlikely.